src/utils/components/SSHAccess/components/SSHCheckbox.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+
+import { ServiceModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachine, V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sDelete } from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, AlertVariant, Checkbox, Flex, FlexItem } from '@patternfly/react-core';
+
+import { createSSHService } from '../utils';
+
+type SSHCheckboxProps = {
+  sshServiceLoaded: boolean;
+  initialSSHService: IoK8sApiCoreV1Service;
+  vmi: V1VirtualMachineInstance;
+  vm: V1VirtualMachine;
+};
+
+const SSHCheckbox: React.FC<SSHCheckboxProps> = ({
+  vm,
+  vmi,
+  initialSSHService,
+  sshServiceLoaded,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const [sshService, setSSHService] = React.useState(initialSSHService);
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState<Error>();
+
+  const onSSHChange = React.useCallback(
+    async (enableSSH: boolean) => {
+      setLoading(true);
+      try {
+        if (enableSSH) {
+          setSSHService((await createSSHService(vm, vmi)) as IoK8sApiCoreV1Service);
```

Comment:
u are saving a promise.. no point in await here..

Reply:
if you arent dealing with the result of await in the same scope it just the same as regular promise

Reply:
maybe createahSSHService.then(()=>setSSHService()) ? can work?

Reply:
just remove the finally and move it also to .then